eras/shelley/impl/src/Cardano/Ledger/Shelley/Rules/Bbody.hs

```diff
@@ -72,6 +80,38 @@ data ShelleyBbodyPredFailure era
   | LedgersFailure (PredicateFailure (EraRule "LEDGERS" era)) -- Subtransition Failures
   deriving (Generic)
 
+type instance EraRuleFailure "BBODY" (ShelleyEra c) = ShelleyBbodyPredFailure (ShelleyEra c)
+
+instance InjectRuleFailure "BBODY" ShelleyBbodyPredFailure (ShelleyEra c) where
+  injectFailure = id
+
+instance InjectRuleFailure "BBODY" ShelleyLedgersPredFailure (ShelleyEra c) where
+  injectFailure = LedgersFailure
+
+instance InjectRuleFailure "BBODY" ShelleyLedgerPredFailure (ShelleyEra c) where
+  injectFailure = LedgersFailure . injectFailure
+
+instance InjectRuleFailure "BBODY" ShelleyUtxowPredFailure (ShelleyEra c) where
+  injectFailure = LedgersFailure . injectFailure
+
+instance InjectRuleFailure "BBODY" ShelleyUtxoPredFailure (ShelleyEra c) where
+  injectFailure = LedgersFailure . injectFailure
+
+instance InjectRuleFailure "BBODY" ShelleyPpupPredFailure (ShelleyEra c) where
+  injectFailure = LedgersFailure . injectFailure
+
+instance InjectRuleFailure "BBODY" ShelleyDelegsPredFailure (ShelleyEra c) where
+  injectFailure = LedgersFailure . injectFailure
+
+instance InjectRuleFailure "BBODY" ShelleyDelplPredFailure (ShelleyEra c) where
+  injectFailure = LedgersFailure . injectFailure
+
+instance InjectRuleFailure "BBODY" ShelleyPoolPredFailure (ShelleyEra c) where
+  injectFailure = LedgersFailure . injectFailure
+
+instance InjectRuleFailure "BBODY" ShelleyDelegPredFailure (ShelleyEra c) where
+  injectFailure = LedgersFailure . injectFailure
+
```

Comment:
Couldn't GHC derive these automatically via transitivity?
```suggestion
instance InjectRuleFailure "LEDGERS" t (ShelleyEra c) => InjectRuleFailure "BBODY" t (ShelleyEra c) where
  injectFailure = LedgersFailure . injectFailure
```

Reply:
As discussed during the meeting, this would result in overlapping instances, which we really want to avoid